packages/react-utils/src/hooks/useTheme.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState, useCallback } from 'react'
+
+const THEME_KEY = 'data-theme'
+const cachedTheme = localStorage.getItem(THEME_KEY)
+
+type Themes = 'light' | 'dark'
+
+export function useTheme(preferredTheme?: Themes) {
+  const [theme, setTheme] = useState(cachedTheme ?? preferredTheme)
```

Comment:
Since this is a condition, it would be better for us to set it using the callback version which is a little more performant in this scenario.

```suggestion
  const [theme, setTheme] = useState(() => cachedTheme ?? preferredTheme)
```